lib/ex_format/lines.ex

```diff
@@ -0,0 +1,15 @@
+defmodule ExFormat.Lines do
+  def initialize_lines_store(string) do
+  	lines = String.split(string, "\n")
+    Agent.start_link(fn -> %{} end, name: :lines)
+    for {line, i} <- Enum.with_index(lines) do
+      update_line(i + 1, String.trim(line))
+    end
```

Comment:
https://github.com/uohzxela/ex_format/blob/4b60582178fb161f5e3603620d09d9ce0d201433/lib/ex_format/lines.ex#L6

So that exact problem is, if you remove `Agent.start_link(fn -> %{} end, name: :lines)` at line 6 and replace the code at line 17 with `Agent.start_link(fn -> lines_map end, name: :lines)`. After you run `mix test`, some of the test cases failed because the agent is still using the state from the previous test.

If I initialized agent with an empty %{}, then the previous state will get cleared before every test, hence it works.

I'm not sure why this is happening. It also happens when I tried running ExFormat in IEx shell, e.g. the agent is still holding the state of the previous version of the file even though I changed the file prior to calling `ExFormat.format`. Any ideas?